src/newcharacter.cpp

```diff
@@ -518,8 +518,8 @@ bool avatar::create( character_type type, const std::string &tempname )
     for( mtype_id elem : prof->pets() ) {
         starting_pets.push_back( elem );
     }
-    if( !prof->vehicle().is_null() && prof->vehicle().is_valid() ) {
-        starting_vehicle = prof->vehicle();
+    if( prof->vehicle() ) {
```

Comment:
`vehicle()` returns a `cata::optional<vproto_id>`, which is the same type as `starting_vehicle `, so you can just assign it directly and don't need to check for the optional containing anything (without the `if`).
```suggestion
    starting_vehicle = prof->vehicle();
```